lib/antiCheat.ts

```diff
@@ -44,6 +45,37 @@ exports.calculateFindItCheatScore = async (challenge: Challenge) => { // TODO Co
   const timestamp = new Date()
   let timeFactor = 0.001
   timeFactor *= (challenge.key === 'scoreBoardChallenge' && config.get('hackingInstructor.isEnabled') ? 0.5 : 1)
+
+  // checks if identicalChallenges is already fetched
+  if (identicalChallenges.length === 0) {
+    identicalChallenges = await retrieveIdenticalChallenges()
+  }
+
+  // identicalChallenges[identicalIndex] will store an array of challenges identical to challenge.key
+  let identicalIndex = -1
+  identicalChallenges.forEach((row, i) => {
+    row.forEach((element) => {
+      if (element === challenge.key) {
+        identicalIndex = i
+      }
+    })
+  })
+  // loop through the challenges that are identical to the current 'challenge' variable and set timeFactor to 0 if one of the identical challenges are in 'find it' phase
+  identicalChallenges[identicalIndex].map(identicalChallenge => {
+    if (identicalChallenge !== challenge.key) {
+      solves.map(element => {
+        if (identicalChallenge === element.challenge.key) {
+          if (element.phase === 'find it') {
+            timeFactor = 0
+            return element
+          }
+        }
+        return element
+      })
```

Comment:
The entire "identical challenge detection" block here should probably be moved into its own function and not happen here in this function.

E.g. to make it clear what is getting checked here and what the acutal outcome is when an identical challenge has already been solved.
```suggestion
  if (wasIdenticalChallengeAlreadySolved(challenge)) {
    timeFactor = 0
  }
```